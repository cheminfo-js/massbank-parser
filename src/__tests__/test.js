import parse from '..';

import fs from 'fs';

let data = fs.readFileSync(`${__dirname}/data.txt`, 'utf8');
describe('test massbank-parser', () => {
  it('should return the same json', () => {
    expect(parse(data)).toMatchSnapshot();
  });
});
