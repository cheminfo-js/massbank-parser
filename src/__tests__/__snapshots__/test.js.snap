// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test massbank-parser should return the same json 1`] = `
Object {
  "accession": "AU100801",
  "acquisition": Object {
    "chromatography": Object {
      "columnName": "Acclaim RSLC C18 2.2um, 2.1x100mm, Thermo",
      "flowGradient": "99/1 at 0-1 min, 61/39 at 3 min, 0.1/99.9 at 14-16 min, 99/1 at 16.1-20 min",
      "flowRate": "200 uL/min at 0-3 min, 400 uL/min at 14 min, 480 uL/min at 16-19 min, 200 uL/min at 19.1-20 min",
      "retentionTime": "4.4 min",
      "solvent": "A water with 0.01% formic acid and 5mM ammonium formateB 90",
    },
    "instrument": "Bruker maXis Impact",
    "instrumentType": "LC-ESI-QTOF",
    "massSpectrometry": Object {
      "collisionEnergy": "Ramp 20.8-31.3 eV",
      "fragmentationMode": "CID",
      "ionMode": "POSITIVE",
      "ionization": "ESI",
      "msType": "MS2",
      "resolution": "35000",
    },
  },
  "authors": "Nikiforos Alygizakis, Anna Bletsou, Nikolaos Thomaidis, University of Athens",
  "chemical": Object {},
  "chemistry": Object {
    "compoundClass": "N/A; Environmental Standard",
    "exactMass": "278.0837",
    "formula": "C12H14N4O2S",
    "iupac": "InChI=1S/C12H14N4O2S/c1-8-7-9(2)15-12(14-8)16-19(17,18)11-5-3-10(13)4-6-11/h3-7H,13H2,1-2H3,(H,14,15,16)",
    "link": Object {
      "cas": "57-68-1",
      "chemspider": "5136",
      "inchikey": "ASWVTGNCAZCNNR-UHFFFAOYSA-N",
      "kegg": "C19530",
      "pubchem": "CID",
    },
    "name": "Sulfadimidine
4-amino-N-(4,6-dimethylpyrimidin-2-yl)benzenesulfonamide",
    "smiles": "Cc1cc(nc(n1)NS(=O)(=O)c2ccc(cc2)N)C",
  },
  "chromatography": Object {},
  "comment": "CONFIDENCE standard compound
INTERNAL_ID 1008",
  "copyright": "Copyright (C) 2015 Department of Chemistry, University of Athens",
  "date": "2015.07.05",
  "license": "CC BY",
  "mass": Object {
    "dataProcessing": "RECALIBRATE identity on assigned fragments and MS1
REANALYZE Peaks with additional N2/O included
WHOLE RMassBank 1.8.1",
    "focusedIon": Object {
      "basePeak": "279.0924",
      "precursorM/Z": "279.091",
      "precursorType": "[M+H]+",
    },
  },
  "peak": Object {
    "annotation": Array [
      Object {
        "error(ppm)": -13.32,
        "formula_count": 1,
        "m/z": 53.0379,
        "mass": 53.0386,
        "tentative_formula": "C4H5+",
      },
      Object {
        "error(ppm)": -6.95,
        "formula_count": 1,
        "m/z": 54.0335,
        "mass": 54.0338,
        "tentative_formula": "C3H4N+",
      },
      Object {
        "error(ppm)": -4.93,
        "formula_count": 2,
        "m/z": 55.0176,
        "mass": 55.0178,
        "tentative_formula": "C3H3O+",
      },
      Object {
        "error(ppm)": -9.63,
        "formula_count": 1,
        "m/z": 65.0379,
        "mass": 65.0386,
        "tentative_formula": "C5H5+",
      },
      Object {
        "error(ppm)": -5.52,
        "formula_count": 1,
        "m/z": 68.0491,
        "mass": 68.0495,
        "tentative_formula": "C4H6N+",
      },
      Object {
        "error(ppm)": -7.98,
        "formula_count": 2,
        "m/z": 69.0329,
        "mass": 69.0335,
        "tentative_formula": "C4H5O+",
      },
      Object {
        "error(ppm)": -5.33,
        "formula_count": 1,
        "m/z": 78.0334,
        "mass": 78.0338,
        "tentative_formula": "C5H4N+",
      },
      Object {
        "error(ppm)": -0.39,
        "formula_count": 3,
        "m/z": 79.0178,
        "mass": 79.0178,
        "tentative_formula": "C5H3O+",
      },
      Object {
        "error(ppm)": -6.69,
        "formula_count": 1,
        "m/z": 80.0489,
        "mass": 80.0495,
        "tentative_formula": "C5H6N+",
      },
      Object {
        "error(ppm)": -4.13,
        "formula_count": 1,
        "m/z": 81.0444,
        "mass": 81.0447,
        "tentative_formula": "C4H5N2+",
      },
      Object {
        "error(ppm)": -4.03,
        "formula_count": 2,
        "m/z": 82.0284,
        "mass": 82.0287,
        "tentative_formula": "C4H4NO+",
      },
      Object {
        "error(ppm)": 4.18,
        "formula_count": 2,
        "m/z": 92.0499,
        "mass": 92.0495,
        "tentative_formula": "C6H6N+",
      },
      Object {
        "error(ppm)": -16.45,
        "formula_count": 2,
        "m/z": 93.0558,
        "mass": 93.0573,
        "tentative_formula": "C6H7N+",
      },
      Object {
        "error(ppm)": -4.42,
        "formula_count": 2,
        "m/z": 94.0647,
        "mass": 94.0651,
        "tentative_formula": "C6H8N+",
      },
      Object {
        "error(ppm)": 4.79,
        "formula_count": 2,
        "m/z": 95.0608,
        "mass": 95.0604,
        "tentative_formula": "C5H7N2+",
      },
      Object {
        "error(ppm)": -0.63,
        "formula_count": 4,
        "m/z": 96.0443,
        "mass": 96.0444,
        "tentative_formula": "C5H6NO+",
      },
      Object {
        "error(ppm)": -2.86,
        "formula_count": 4,
        "m/z": 108.0461,
        "mass": 108.0464,
        "tentative_formula": "CH8N4S+",
      },
      Object {
        "error(ppm)": -10.66,
        "formula_count": 2,
        "m/z": 109.0497,
        "mass": 109.0509,
        "tentative_formula": "C4H5N4+",
      },
      Object {
        "error(ppm)": -3.99,
        "formula_count": 4,
        "m/z": 110.0616,
        "mass": 110.0621,
        "tentative_formula": "CH10N4S+",
      },
      Object {
        "error(ppm)": -12.63,
        "formula_count": 2,
        "m/z": 111.0651,
        "mass": 111.0665,
        "tentative_formula": "C4H7N4+",
      },
      Object {
        "error(ppm)": -3.88,
        "formula_count": 4,
        "m/z": 120.0565,
        "mass": 120.057,
        "tentative_formula": "C8H8O+",
      },
      Object {
        "error(ppm)": 3,
        "formula_count": 4,
        "m/z": 122.0716,
        "mass": 122.0713,
        "tentative_formula": "C6H8N3+",
      },
      Object {
        "error(ppm)": 2.04,
        "formula_count": 4,
        "m/z": 123.0794,
        "mass": 123.0791,
        "tentative_formula": "C6H9N3+",
      },
      Object {
        "error(ppm)": 2.39,
        "formula_count": 3,
        "m/z": 124.0872,
        "mass": 124.0869,
        "tentative_formula": "C6H10N3+",
      },
      Object {
        "error(ppm)": 28.57,
        "formula_count": 3,
        "m/z": 125.0905,
        "mass": 125.0869,
        "tentative_formula": "C4H15NOS+",
      },
      Object {
        "error(ppm)": 0.89,
        "formula_count": 4,
        "m/z": 126.0663,
        "mass": 126.0662,
        "tentative_formula": "C5H8N3O+",
      },
      Object {
        "error(ppm)": 28.24,
        "formula_count": 3,
        "m/z": 127.0697,
        "mass": 127.0662,
        "tentative_formula": "C3H13NO2S+",
      },
      Object {
        "error(ppm)": 2.21,
        "formula_count": 4,
        "m/z": 156.0117,
        "mass": 156.0114,
        "tentative_formula": "C6H6NO2S+",
      },
      Object {
        "error(ppm)": 1.99,
        "formula_count": 5,
        "m/z": 157.0148,
        "mass": 157.0145,
        "tentative_formula": "C7HN4O+",
      },
      Object {
        "error(ppm)": 8.57,
        "formula_count": 4,
        "m/z": 158.0072,
        "mass": 158.0059,
        "tentative_formula": "C9H4NS+",
      },
      Object {
        "error(ppm)": 6.61,
        "formula_count": 4,
        "m/z": 174.0224,
        "mass": 174.0212,
        "tentative_formula": "C12H2N2+",
      },
      Object {
        "error(ppm)": 1.38,
        "formula_count": 3,
        "m/z": 186.0334,
        "mass": 186.0332,
        "tentative_formula": "C6H8N3O2S+",
      },
      Object {
        "error(ppm)": -4.37,
        "formula_count": 3,
        "m/z": 187.0368,
        "mass": 187.0376,
        "tentative_formula": "C9H5N3O2+",
      },
      Object {
        "error(ppm)": 5.44,
        "formula_count": 3,
        "m/z": 188.0128,
        "mass": 188.0117,
        "tentative_formula": "C11N4+",
      },
      Object {
        "error(ppm)": 0.39,
        "formula_count": 5,
        "m/z": 188.0291,
        "mass": 188.029,
        "tentative_formula": "C11H8OS+",
      },
      Object {
        "error(ppm)": 7.17,
        "formula_count": 3,
        "m/z": 204.0445,
        "mass": 204.043,
        "tentative_formula": "C12H4N4+",
      },
      Object {
        "error(ppm)": -17.42,
        "formula_count": 3,
        "m/z": 205.0473,
        "mass": 205.0509,
        "tentative_formula": "C12H5N4+",
      },
      Object {
        "error(ppm)": 4.94,
        "formula_count": 3,
        "m/z": 206.0406,
        "mass": 206.0396,
        "tentative_formula": "C11H10O2S+",
      },
      Object {
        "error(ppm)": 2.94,
        "formula_count": 1,
        "m/z": 213.1141,
        "mass": 213.1135,
        "tentative_formula": "C12H13N4+",
      },
      Object {
        "error(ppm)": -21.61,
        "formula_count": 1,
        "m/z": 214.1167,
        "mass": 214.1213,
        "tentative_formula": "C12H14N4+",
      },
      Object {
        "error(ppm)": 0.01,
        "formula_count": 3,
        "m/z": 215.0927,
        "mass": 215.0927,
        "tentative_formula": "C11H11N4O+",
      },
      Object {
        "error(ppm)": -0.06,
        "formula_count": 1,
        "m/z": 215.1291,
        "mass": 215.1291,
        "tentative_formula": "C12H15N4+",
      },
      Object {
        "error(ppm)": 5.22,
        "formula_count": 1,
        "m/z": 279.0925,
        "mass": 279.091,
        "tentative_formula": "C12H15N4O2S+",
      },
      Object {
        "error(ppm)": 32.3,
        "formula_count": 1,
        "m/z": 280.0953,
        "mass": 280.0863,
        "tentative_formula": "C11H14N5O2S+",
      },
      Object {
        "error(ppm)": 7.91,
        "formula_count": 3,
        "m/z": 281.0725,
        "mass": 281.0703,
        "tentative_formula": "C11H13N4O3S+",
      },
      Object {
        "error(ppm)": -13.87,
        "formula_count": 2,
        "m/z": 282.0742,
        "mass": 282.0781,
        "tentative_formula": "C11H14N4O3S+",
      },
    ],
    "numPeak": "46",
    "peak": Array [
      Object {
        "int.": 1232,
        "m/z": 53.0379,
        "rel.int.": 8,
      },
      Object {
        "int.": 912,
        "m/z": 54.0335,
        "rel.int.": 6,
      },
      Object {
        "int.": 824,
        "m/z": 55.0176,
        "rel.int.": 5,
      },
      Object {
        "int.": 12012,
        "m/z": 65.0379,
        "rel.int.": 87,
      },
      Object {
        "int.": 17932,
        "m/z": 68.0491,
        "rel.int.": 130,
      },
      Object {
        "int.": 2260,
        "m/z": 69.0329,
        "rel.int.": 16,
      },
      Object {
        "int.": 2036,
        "m/z": 78.0334,
        "rel.int.": 14,
      },
      Object {
        "int.": 3116,
        "m/z": 79.0178,
        "rel.int.": 22,
      },
      Object {
        "int.": 1972,
        "m/z": 80.0489,
        "rel.int.": 14,
      },
      Object {
        "int.": 2688,
        "m/z": 81.0444,
        "rel.int.": 19,
      },
      Object {
        "int.": 836,
        "m/z": 82.0284,
        "rel.int.": 6,
      },
      Object {
        "int.": 42144,
        "m/z": 92.0499,
        "rel.int.": 305,
      },
      Object {
        "int.": 3920,
        "m/z": 93.0558,
        "rel.int.": 28,
      },
      Object {
        "int.": 2324,
        "m/z": 94.0647,
        "rel.int.": 16,
      },
      Object {
        "int.": 4172,
        "m/z": 95.0608,
        "rel.int.": 30,
      },
      Object {
        "int.": 1792,
        "m/z": 96.0443,
        "rel.int.": 12,
      },
      Object {
        "int.": 46776,
        "m/z": 108.0461,
        "rel.int.": 339,
      },
      Object {
        "int.": 3252,
        "m/z": 109.0497,
        "rel.int.": 23,
      },
      Object {
        "int.": 8416,
        "m/z": 110.0616,
        "rel.int.": 61,
      },
      Object {
        "int.": 716,
        "m/z": 111.0651,
        "rel.int.": 5,
      },
      Object {
        "int.": 2704,
        "m/z": 120.0565,
        "rel.int.": 19,
      },
      Object {
        "int.": 8376,
        "m/z": 122.0716,
        "rel.int.": 60,
      },
      Object {
        "int.": 3096,
        "m/z": 123.0794,
        "rel.int.": 22,
      },
      Object {
        "int.": 98124,
        "m/z": 124.0872,
        "rel.int.": 711,
      },
      Object {
        "int.": 8816,
        "m/z": 125.0905,
        "rel.int.": 63,
      },
      Object {
        "int.": 24680,
        "m/z": 126.0663,
        "rel.int.": 178,
      },
      Object {
        "int.": 820,
        "m/z": 127.0697,
        "rel.int.": 5,
      },
      Object {
        "int.": 114168,
        "m/z": 156.0117,
        "rel.int.": 827,
      },
      Object {
        "int.": 7908,
        "m/z": 157.0148,
        "rel.int.": 57,
      },
      Object {
        "int.": 2128,
        "m/z": 158.0072,
        "rel.int.": 15,
      },
      Object {
        "int.": 1524,
        "m/z": 174.0224,
        "rel.int.": 11,
      },
      Object {
        "int.": 15520,
        "m/z": 186.0334,
        "rel.int.": 112,
      },
      Object {
        "int.": 1068,
        "m/z": 187.0368,
        "rel.int.": 7,
      },
      Object {
        "int.": 2256,
        "m/z": 188.0128,
        "rel.int.": 16,
      },
      Object {
        "int.": 736,
        "m/z": 188.0291,
        "rel.int.": 5,
      },
      Object {
        "int.": 137792,
        "m/z": 204.0445,
        "rel.int.": 999,
      },
      Object {
        "int.": 9608,
        "m/z": 205.0473,
        "rel.int.": 69,
      },
      Object {
        "int.": 4628,
        "m/z": 206.0406,
        "rel.int.": 33,
      },
      Object {
        "int.": 25160,
        "m/z": 213.1141,
        "rel.int.": 182,
      },
      Object {
        "int.": 3088,
        "m/z": 214.1167,
        "rel.int.": 22,
      },
      Object {
        "int.": 4232,
        "m/z": 215.0927,
        "rel.int.": 30,
      },
      Object {
        "int.": 1820,
        "m/z": 215.1291,
        "rel.int.": 13,
      },
      Object {
        "int.": 84720,
        "m/z": 279.0925,
        "rel.int.": 614,
      },
      Object {
        "int.": 11628,
        "m/z": 280.0953,
        "rel.int.": 84,
      },
      Object {
        "int.": 10800,
        "m/z": 281.0725,
        "rel.int.": 78,
      },
      Object {
        "int.": 1684,
        "m/z": 282.0742,
        "rel.int.": 12,
      },
    ],
    "splash": "splash10-0pk9-2970000000-abb9e31dc053c1d21884",
  },
  "publication": "",
  "recordTitle": "Sulfadimidine; LC-ESI-QTOF; MS2; CE",
}
`;
